### ISSUE TITLE
Python-eccodes version numbering changed again, Travis is testing against an old version

### ISSUE THREAD

**Issue by stephenworsley** [SciTools/iris#3723]
Python-eccodes previously changed their version numbering convention which resulted in #3593 being necessary in order to keep Travis using the latest version. Recently, python-eccodes has changed their version numbering convention again (see https://anaconda.org/conda-forge/python-eccodes/files) so that the changes made in #3593 are now causing Travis to not install the latest version. In order to pick up the latest version again, the changes made in #3593 should be reverted.

**Comment by pp-mo** [comment]
AFAICT the problem here is *not* that the numbering system has changed.
It is that they have provided 2 builds for the _same_ "version" (2020.04.0) without incrementing the build number.  Which is simply a thing you must **_never_** do in a conda channel !!

E.G. the file "linux-64/python-eccodes-2020.04.0-py36hda16e5c_0.tar.bz2" dates to 2020-05-04 @14:25
And its lib/info/recipe.yaml has
```
package:
    name: python-eccodes
    version: 2020.04.0
source:
    sha256: b2589c1850d0906d5b22bf699c0977736a704ee35799e4ea618e253332793ac4
    url: https://github.com/ecmwf/eccodes-python/archive/2020.04.0.tar.gz
build:
    number: '0'
    string: py38h8790de6_0
```

The file "linux-64/python-eccodes-2020.04.0-py38h8790de6_0.tar.bz2" dates to 2020-05-04 @14:29
(5 minutes later !)
And it provides colliding info
```
package:
    name: python-eccodes
    version: 2020.04.0
source:
    sha256: b2589c1850d0906d5b22bf699c0977736a704ee35799e4ea618e253332793ac4
    url: https://github.com/ecmwf/eccodes-python/archive/2020.04.0.tar.gz
build:
    number: '0'
    string: py36hda16e5c_0
```

**Comment by trexfeathers** [comment]
While the 2 identical builds is doubtless a very bad thing to do, we shouldn't be encountering this issue at all while Travis is still pinned to the following:

`python-eccodes">=0.9.1, <2"`

Version `2020` is most definitely `>2`!

**Comment by pp-mo** [comment]
> Version 2020 is most definitely >2!

I don't really understand that aspect.
It also appears, from comparing the two meta.yaml files, that they are nailing tight many dependencies, so one of the above is fixed to python 3.6, and the other 3.8.

**Comment by pp-mo** [comment]
Ok, I suspect this *could* be "what I just said", and not the version spec in travis.yml.
The problem is, we don't integrate with the original environment resolution, but install eccodes in a secondary step.  
So, I think the eccodes "meta.yaml" for the python 3.6 version now requires **_exactly_** "python=3.6.9", 
whereas we are trying to install in the environment we just created, which has 3.6.10. 
According to https://travis-ci.org/github/SciTools/iris/jobs/691806720 

So it might be installing the earlier one because it has a looser Python version requirement ??
I'll just check that out ...

**Comment by pp-mo** [comment]
Update : removing the pin [gives segfaults](https://github.com/SciTools/iris/pull/3724#issuecomment-637546516)
Maybe we need to pause here, and schedule work to address it at a better time :-(

**Comment by pp-mo** [comment]
> removing the pin gives segfaults
Until a new iris-grib, which will fix it : SciTools/iris-grib#208

**Comment by lbdreyer** [comment]
We no longer install eccodes into Travis, so this can be closed
